compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -1055,6 +1055,15 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
         let mut llargs = Vec::with_capacity(arg_count);
 
+        if kind == CallKind::Tail && fn_abi.ret.is_indirect() {
+            match self.make_return_dest(bx, destination, &fn_abi.ret, &mut llargs) {
+                ReturnDest::Nothing => {}
+                _ => bug!(
+                    "tail calls to functions with indirect returns cannot store into a destination"
+                ),
```

Comment:
I think this is correct. `make_return_dest` only returns something other than `ReturnDest::Nothing` for indirect returns when the destination is an `LocalRef::PendingOperand`, but `_0` should be a `LocalRef::Place` for indirect returns and tail calls shouldn't use any destination other than `_0`.